src/providers/BackgroundBlurProvider/index.tsx

```diff
@@ -0,0 +1,147 @@
+  NoOpVideoFrameProcessor,
+  VideoFrameProcessor,
+} from 'amazon-chime-sdk-js';
+import React, {
+  createContext,
+  FC,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+interface Props {
+  /** The spec defines the assets that will be used for adding background blur to a frame */
+  spec?: BackgroundFilterSpec;
+  /** A set of options that can be supplied when creating a background blur video frame processor. */
+  options?: BackgroundBlurOptions;
+}
+
+interface BackgroundBlurProviderState {
+  createBackgroundBlurDevice: (
+    device: Device
+  ) => Promise<DefaultVideoTransformDevice>;
+  isBackgroundBlurSupported: boolean | undefined;
+}
+
+const BackgroundBlurProviderContext =
+  createContext<BackgroundBlurProviderState | null>(null);
+
+const BackgroundBlurProvider: FC<Props> = ({ spec, options, children }) => {
+  const [isBackgroundBlurSupported, setIsBackgroundBlurSupported] = useState<
+    boolean | undefined
+  >(undefined);
+  const [processor, setProcessor] = useState<VideoFrameProcessor | undefined>();
+
+  useEffect(() => {
+    async function initializeBackgroundBlur() {
+      try {
+        console.log('Initializing background blur processor.');
+        const createdProcessor = await BackgroundBlurVideoFrameProcessor.create(
+          spec,
+          options
+        );
+
+        if (createdProcessor instanceof NoOpVideoFrameProcessor) {
+          console.log(`Initialized NoOpVideoFrameProcessor.`);
+          setProcessor(undefined);
+          setIsBackgroundBlurSupported(false);
+        } else {
+          console.log(
+            `Initialized background blur processor: ${JSON.stringify(
+              processor
+            )}`
+          );
+          setProcessor(createdProcessor);
+          setIsBackgroundBlurSupported(true);
+        }
+      } catch (error) {
+        console.error(
+          `Error creating a background blur video frame processor device.`,
+          error
+        );
+        setProcessor(undefined);
+        setIsBackgroundBlurSupported(false);
+      }
+    }
+    initializeBackgroundBlur();
+    return () => {
+      console.log(`Destroying background blur processor.`);
+      processor?.destroy();
+    };
+  }, []);
+
+  const createBackgroundBlurDevice = async (
+    selectedDevice: Device
+  ): Promise<DefaultVideoTransformDevice> => {
+    console.log(
+      `Calling createBackgroundBlurDevice with device: ${selectedDevice}`
+    );
+    // It takes time for the processor to start - if you call this method before it is finished initializing, throw an error
+    if (!isBackgroundBlurSupported) {
+      throw new Error(
+        'Background blur is not supported. The processor may not be initialized yet.'
+      );
+    }
+    try {
+      const logger = options?.logger
+        ? options.logger
+        : new ConsoleLogger('BackgroundBlurProvider', LogLevel.INFO);
+      if (processor) {
+        const chosenVideoTransformDevice = new DefaultVideoTransformDevice(
```

Comment:
I though we were abstracting the `DefaultVideoTransformDevice` creation out from `BackgroundBlurProvider`.
Cause, if there are multiple processors to apply will we create multiple `DefaultVideoTransformDevice` with a single processor through the processors array and will each of them have its own provider? Or, we wont support such a use case and if background blur is enabled builders wont be able to use other processors.

Reply:
@devalevenkatesh I think we had a meeting about this and decided we only supported blur provider for now so there would be no multi processors use case.